dbms/src/Columns/ColumnString.h

```diff
@@ -247,6 +247,48 @@ class ColumnString final : public COWPtrHelper<IColumn, ColumnString>
         return pos + string_size;
     }
 
+    void countSerializeByteSize(PaddedPODArray<size_t> & byte_size) const override
+    {
+        if unlikely (byte_size.size() != offsets.size())
+            byte_size.resize(offsets.size());
+
+        size_t size = byte_size.size();
+        for (size_t i = 0; i < size; ++i)
+            byte_size[i] += sizeof(size_t) + sizeAt(i);
+    }
+
+    void serializeToPos(PaddedPODArray<UInt8 *> & pos, size_t start, size_t end, bool has_null) const override
+    {
+        if (has_null)
+            serializeToPosImpl<true>(pos, start, end);
+        else
+            serializeToPosImpl<false>(pos, start, end);
+    }
+
+    template <bool has_null>
+    void serializeToPosImpl(PaddedPODArray<UInt8 *> & pos, size_t start, size_t end) const
+    {
+        if unlikely (pos.size() != offsets.size())
```

Comment:
when will this happen?

Reply:
It should not happen. Maybe it can be changed to the exception?